core/frontend/src/RealityDataProviderRegistry.ts

```diff
@@ -0,0 +1,49 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+/** @packageDocumentation
+ * @module Tiles
+ */
+
+/** Returns the URL used for retrieving Google Photorealistic 3D Tiles.
+ * @alpha
+ */
+export function getGooglePhotorealistic3DTilesURL() {
+  return "https://tile.googleapis.com/v1/3dtiles/root.json";
+}
+
+/** Stores key-value pair to be added to reality data requests.
+ * @alpha
+ */
+export interface RealityDataProviderKey {
+  key: string;
+  value: string;
+}
+
+/** Options supplied at startup via [[IModelAppOptions.realityDataOptions]] to specify access keys for various reality data formats.
+ * `gp3dt` must have its key value set to `key`
+ * @alpha
+ */
+export interface RealityDataProviderOptions {
+  /** Access key for Google Photorealistic 3D Tiles in the format `{ key: "key", value: "your-gp3dt-key" }`. */
+  gp3dt?: RealityDataProviderKey;
```

Comment:
should we be hardcoding these in core-frontend?
also take into account those who will not be providing an access token/key, but has to interface with a service to obtain a session token. you should look at what was done for google 2d tiles

Reply:
Based on earlier feedback (and now your feedback as well), we are in the process of changing the way this works. New approach:

1. User can create a GP3DT provider (new API class)
2. User can register that provider themselves if they want to use it.
3. User will be responsible for passing in a valid key, if desired.
4. An alternate using a session token / [authorization header](https://github.com/iTwin/itwinjs-core/pull/8104#discussion_r2103521100) will be available - but we need to develop this.

Stay tuned for further code updates in these areas (the first three of these are done; we need to clean up old classes still).